crates/assists/src/utils.rs

```diff
@@ -179,6 +179,32 @@ fn invert_special_case(expr: &ast::Expr) -> Option<ast::Expr> {
             ast::BinOp::EqualityTest => bin.replace_op(T![!=]).map(|it| it.into()),
             _ => None,
         },
+        ast::Expr::MethodCallExpr(mce) => {
+            const IS_SOME_TEXT: &str = "is_some";
+            const IS_NONE_TEXT: &str = "is_none";
+            const IS_OK_TEXT: &str = "is_ok";
+            const IS_ERR_TEXT: &str = "is_err";
+
+            let name = mce.name_ref()?;
+            let name_text = name.text();
+            if !(name_text != IS_SOME_TEXT
+                || name_text != IS_NONE_TEXT
+                || name_text != IS_OK_TEXT
+                || name_text != IS_ERR_TEXT)
+            {
+                return None;
+            }
```

Comment:
Can yoy remove this if, and then make `caller` a closure (`|| &mce...`) that you call where you are currently using the `caller` value.